### ISSUE TITLE
`redundant-attachment` lint misinterprets declared `ρ` causing false positives in `eo-runtime`

### ISSUE THREAD

**Issue by maxonfjvipon** [objectionary/lints#1301]
Third one from making the receiver explicit in `eo-runtime`, after #1299 and #1300. This one is subtler, because the lint is reasoning correctly from a premise that stops holding.

`redundant-attachment` reports an auto-named object when its name is unreferenced and `eo:outer-refs` is empty:

```xml
<xsl:if test="empty(eo:outer-refs(.)) and (count($refs)=0 or ...)">
```

The comment explains why the second half matters: dropping the name inlines the object, and a `ξ.ρ.`-based reference reaching past its own voids into the enclosing scope would then be caught by `anonymous-formation` instead.

Once the head declares `^`, `ρ` is one of the object's own voids. A `ξ.ρ.` reference no longer reaches past them, so `outer-refs` comes back empty for formations that plainly do reach outward, and thirteen of them in `eo-runtime` get flagged. In `while.eo`:

```
^.^.^.max.gt ^.iter > [^ i] >>
```

That body reads `max` and `iter` from two different enclosing scopes. Nothing about it became self-contained; only the spelling of the receiver changed.

Dropping the suffix does satisfy the lint, but the formatter then rewrites the call site into vertical-argument form, and that reshaping changed the `ρ` chains enough to break `map`, `while`, `range` and `input` when we tried it.

Treat `ρ` as reaching outward even when declared — it is the receiver, and its value comes from the dispatch site rather than from inside the object. The distinction the rule wants is between references satisfied by the object's own parameters and references satisfied by its context, and `ρ` has always been the latter.

**Comment by 0crat** [comment]
@maxonfjvipon Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+1013](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
